### ISSUE TITLE
Tauri 2.0

### ISSUE THREAD

**Issue by martpie** [pevers/tauri-plugin-nosleep#5]
Hi, would it be possible to update this plugin the support higher versions of Tauri? :)

```
error: failed to select a version for `webkit2gtk-sys`.
    ... required by package `wry v0.15.0`
    ... which satisfies dependency `wry = "^0.15"` of package `tauri-runtime-wry v0.4.0`
    ... which satisfies dependency `tauri-runtime-wry = "^0.4.0"` of package `tauri v1.0.0-rc.8`
    ... which satisfies dependency `tauri = "^1.0.0-rc.8"` of package `tauri-plugin-nosleep v0.1.0`
    ... which satisfies dependency `tauri-plugin-nosleep = "^0.1.0"` of package `museeks v0.20.0 (/Users/martpie/dev/museeks/src-tauri)`
versions that meet the requirements `^0.17` are: 0.17.0

the package `webkit2gtk-sys` links to the native library `web_kit2`, but it conflicts with a previous package which links to `web_kit2` as well:
package `webkit2gtk-sys v2.0.1`
    ... which satisfies dependency `ffi = "^2.0.1"` of package `webkit2gtk v2.0.1`
    ... which satisfies dependency `webkit2gtk = "=2.0.1"` of package `tauri v2.0.0-beta.6`
    ... which satisfies dependency `tauri = "^2.0.0-beta.5"` of package `museeks v0.20.0 (/Users/martpie/dev/museeks/src-tauri)`
Only one package in the dependency graph may specify the same links value. This helps ensure that only one copy of a native library is linked in the final binary. Try to adjust your dependencies so that only one package uses the `links = "web_kit2"` value. For more information, see https://doc.rust-lang.org/cargo/reference/resolver.html#links.

failed to select a version for `webkit2gtk-sys` which could resolve this conflict
error Command failed with exit code 101.
```

**Comment by pevers** [comment]
> Hi, would it be possible to update this plugin the support higher versions of Tauri? :)
> 
> ```
> error: failed to select a version for `webkit2gtk-sys`.
>     ... required by package `wry v0.15.0`
>     ... which satisfies dependency `wry = "^0.15"` of package `tauri-runtime-wry v0.4.0`
>     ... which satisfies dependency `tauri-runtime-wry = "^0.4.0"` of package `tauri v1.0.0-rc.8`
>     ... which satisfies dependency `tauri = "^1.0.0-rc.8"` of package `tauri-plugin-nosleep v0.1.0`
>     ... which satisfies dependency `tauri-plugin-nosleep = "^0.1.0"` of package `museeks v0.20.0 (/Users/martpie/dev/museeks/src-tauri)`
> versions that meet the requirements `^0.17` are: 0.17.0
> 
> the package `webkit2gtk-sys` links to the native library `web_kit2`, but it conflicts with a previous package which links to `web_kit2` as well:
> package `webkit2gtk-sys v2.0.1`
>     ... which satisfies dependency `ffi = "^2.0.1"` of package `webkit2gtk v2.0.1`
>     ... which satisfies dependency `webkit2gtk = "=2.0.1"` of package `tauri v2.0.0-beta.6`
>     ... which satisfies dependency `tauri = "^2.0.0-beta.5"` of package `museeks v0.20.0 (/Users/martpie/dev/museeks/src-tauri)`
> Only one package in the dependency graph may specify the same links value. This helps ensure that only one copy of a native library is linked in the final binary. Try to adjust your dependencies so that only one package uses the `links = "web_kit2"` value. For more information, see https://doc.rust-lang.org/cargo/reference/resolver.html#links.
> 
> failed to select a version for `webkit2gtk-sys` which could resolve this conflict
> error Command failed with exit code 101.
> ```

Yes I'll have a look to upgrade it to Tauri 2.0. Thanks for letting me know

**Comment by pevers** [comment]
I've made a v2 version in the `v2` branch which supports Tauri v2. Beta release is here https://github.com/pevers/tauri-plugin-nosleep/releases/tag/2.0.0-beta1 and here https://crates.io/crates/tauri-plugin-nosleep/versions . This should follow the Tauri SemVer. 

I tested it with an example using v2 and that seems to work.
